packages/ui-common/src/components/RoomPlayer/PlayAgainMenu.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+import {
+  Paper, Stack, Typography, Button,
+  CircularProgress,
+} from '@mui/material';
+import { RoomStartContext } from '@urturn/types-common';
+import logger from '../../logger';
+
+interface PlayAgainMenuProps {
+  spectating: boolean
+  roomStartContext: RoomStartContext
+  playAgain: () => Promise<void>
+}
+
+function PlayAgainMenu({
+  spectating,
+  roomStartContext, playAgain,
+}: PlayAgainMenuProps): React.ReactElement {
+  const [loadingRoom, setLoadingRoom] = useState(false);
+  const buttonMessage = roomStartContext.private ? 'Restart Private Room' : 'Play Again';
+  const disabled = loadingRoom || (roomStartContext.private && spectating);
```

Comment:
why is it only if ur spectating a private room?

Reply:
good question, ill make a comment in the code. But to summarize: `restart private room` button should be disabled if the player is spectating. Only players that are in the `private` room should be allowed to restart the private room.